python/cudf/cudf/core/dataframe.py

```diff
@@ -7133,12 +7133,12 @@ def eval(self, expr: str, inplace: bool = False, **kwargs):
             )
 
         # Have to use a regex match to avoid capturing "=="
-        includes_assignment = re.search("[^=]=[^=]", expr) is not None
+        includes_assignment = re.search("[^=><]=[^=><]", expr) is not None
```

Comment:
I’m a little puzzled. Why is the second part of the expression altered? I don’t think you need to catch =< or => here since those aren’t legal gte/lte operators as far as I know.
```suggestion
        includes_assignment = re.search("[^=><]=[^=]", expr) is not None
```

Reply:
You're right, this was just from a quick find-and-replace.